### PR TITLE
new syntax for duckduckgo + mac-keyboard + gnome-keyboard

### PR DESCRIPTION
**Pull request by RomainLG**
This furthers the work on #1250 

Except those three files I'd say almost none complies with the perfect by-the-book syntax...

### CODE REVIEW

File: share/goodie/cheat_sheets/mac-keyboard.json [hunk 1]
```diff
@@ -9,148 +9,148 @@
...
-            "key" : "[Cmd] + [Shift] + [Z]",
+            "key" : "[Cmd] [Shift] [Z]",
             "val" : "Redo"
         }, {
-            "key" : "[Cmd] + [←/→]",
+            "key" : "[Cmd] [←/→]",
             "val" : "Go to beginning/end of line"
         }, {
-            "key" : "[Cmd] + [↑]",
+            "key" : "[Cmd] [←]",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/mac-keyboard.json]: Looks like this should be `↑`?

File: share/goodie/cheat_sheets/mac-keyboard.json [hunk 2]
```diff
@@ -9,148 +9,148 @@
...
             "val" : "Go to beginning of document"
         }, {
-            "key" : "[Cmd] + [↓]",
+            "key" : "[Cmd] [→]",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/mac-keyboard.json]: and this should be `↓`?

File: share/goodie/cheat_sheets/mac-keyboard.json [hunk 3]
```diff
@@ -9,148 +9,148 @@
...
-            "key" : "[Cmd] + [M]",
+            "key" : "[Cmd] [M]",
             "val" : "Minimize windows of current app"
         }, {
-            "key" : "[Cmd] + [,]",
+            "key" : "[Cmd] [,]",
             "val" : "Open Preferences (if available)"
         }, {
-            "key" : "[Ctrl] + [←/→]",
+            "key" : "[Ctrl] [←/→]",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/mac-keyboard.json]: Should be: `[Ctrl] ([←]/[→])`

File: share/goodie/cheat_sheets/mac-keyboard.json [hunk 4]
```diff
@@ -9,148 +9,148 @@
...
         }, {
             "key" : "Enter",
             "val" : "Open selected item"
         }, {
-            "key" : "[Cmd] + [Enter]",
+            "key" : "[Cmd] [Enter]",
             "val" : "Open selected item in Finder"
         }, {
-            "key" : "[Cmd] + [↑/↓]",
+            "key" : "[Cmd] [↑/↓]",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/mac-keyboard.json]: Should be: `[Cmd] ([↑]/[↓])`

File: share/goodie/cheat_sheets/gnome-keyboard.json [hunk 5]
```diff
@@ -10,66 +10,66 @@
...
-            "key": "[Ctrl] + [Alt] + [Tab] "
+            "key": "[Ctrl] [Alt] [Tab] "
         }, {
             "val": "Show the list of applications",
-            "key": "[Super] + [A] "
+            "key": "[Super] [A] "
         }, {
             "val": "Switch between workspaces",
-            "key": "[Super] + [Page Up] \/ [Page Down]"
+            "key": "[Super] [Page Up] \/ [Page Down]"
```
Comment (moollaza) [inline share/goodie/cheat_sheets/gnome-keyboard.json]: Should be: `[Super] ([Page Up]/[Page Down])

File: share/goodie/cheat_sheets/gnome-keyboard.json [hunk 6]
```diff
@@ -10,66 +10,66 @@
...
         }, {
             "val": "Move the current window to a different workspace",
-            "key": "[Super] + [Shift]+[Page Up] \/ [Page Down]"
+            "key": "[Super] [Shift]+[Page Up] \/ [Page Down]"
```
Comment (moollaza) [inline share/goodie/cheat_sheets/gnome-keyboard.json]: Should be: `[Super] [Shift] ([Page Up]/[Page Down])`

File: share/goodie/cheat_sheets/duckduckgo.json [hunk 7]
```diff
@@ -8,11 +8,11 @@
     "sections": {
         "Open results": [
             {
-                "key": "[↵] or [l] or [o]", 
+                "key": "([↵]/[l]/[o])",
```
Comment (jagtalon) [inline share/goodie/cheat_sheets/duckduckgo.json]: @moollaza looks like this doesn't render properly when there are no spaces in between.

Comment (moollaza) [inline share/goodie/cheat_sheets/duckduckgo.json]: Yes, that's a known issue. This should really be switched to 3 different items, with the other two indicating they're alternatives:

```
{
  key: "...",
  value: "..."
},
{
  key: "...",
  value: "... (alternative)"
},
{
  key: "...",
  value: "... (alternative)"
}
```

I plan to update the parser later, but don't right now.

**Comment by moollaza** [comment]
@RomainLG it looks like there's some merge conflicts with this branch now -- can you please fix them so we can merge this in?

**Comment by RomainLG** [comment]
I must admit I don't understand what is going on with this :s

**Comment by moollaza** [comment]
@RomainLG 
sorry for the delay here. I believe these files have changed, some of them may have had the name changed during cleanup I did and so you've run into conflicts. At this point it may be simplest to create a new branch, pull in the upstream changes, and fix what remains to be fixed in the repo currently

**Pull request closed without merging**